src/messaging/ExchangeContext.cpp

```diff
@@ -306,7 +306,9 @@ CHIP_ERROR ExchangeContext::StartResponseTimer()
         return CHIP_ERROR_INTERNAL;
     }
 
-    return lSystemLayer->StartTimer(mResponseTimeout, HandleResponseTimeout, this);
+    System::Timer * lTimer = lSystemLayer->StartTimer(mResponseTimeout, HandleResponseTimeout, this);
+
+    return (lTimer != nullptr) ? CHIP_NO_ERROR : CHIP_ERROR_INTERNAL;
```

Comment:
consider creating a separate error or returning 'NO_MEMORY' (ideally I would consider this NO_RESOURCES if going for a generic code). This 'ERROR_INTERNAL' is very catch-all and if I see a log saying 'X failed: internal error' that is a really painful debugging session.

Reply:
Ack, I prefer a explicit Error code to indicate start timer failed since NO_RESOURCES is also generic